proxy/http/HttpTransact.h

```diff
@@ -905,8 +905,9 @@ class HttpTransact
     int next_hop_scheme; // out
     int orig_scheme;     // pre-mapped scheme
     int method;
-    int cause_of_death_errno; // in
-    HostDBInfo host_db_info;  // in
+    int cause_of_death_errno;     // in
+    Ptr<HostDBInfo> hostdb_entry; // Pointer to the entry we are referencing in hostdb-- to keep our ref
```

Comment:
Although `host_db_info` is a copy-- none of the related structures are (meaning things that are offsets like rr() etc. aren't copied over. IMO we should probably change this to pointer anyways-- so we can still point directly to the hostdbinfo we want-- without having to do the copy (since it isn't sufficient anyways).